x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/ReindexDataStreamAction.java

```diff
@@ -70,22 +81,55 @@ public boolean equals(Object other) {
 
     }
 
-    public static class ReindexDataStreamRequest extends ActionRequest {
+    public static class ReindexDataStreamRequest extends ActionRequest implements IndicesRequest, ToXContent {
+        private final String mode;
```

Comment:
Can we make the mode an enum for this?

Reply:
I did that and then undid it because I was worried about deserialization failures when we add modes in the future. But maybe failing fast and/or dealing with that with a transport version check is the way to go.